crates/aggkit-prover/src/main.rs

```diff
@@ -1,46 +1,47 @@
-use prover_engine::ProverEngine;
-use tokio_util::sync::CancellationToken;
-use tracing::info;
+use aggkit_prover::version;
+use anyhow::Context as _;
+use clap::Parser as _;
+use sp1_sdk::HashableKey as _;
+use sp1_zkvm::lib::utils::words_to_bytes_le;
 
 fn main() -> anyhow::Result<()> {
-    let config = Arc::new(aggkit_prover_config::ProverConfig::default());
-
-    // Initialize the logger
-    prover_logger::tracing(&config.log);
-
-    let global_cancellation_token = CancellationToken::new();
-
-    info!("Starting AggKit Prover on {}", config.grpc_endpoint);
-
-    let prover_runtime = tokio::runtime::Builder::new_multi_thread()
-        .thread_name("aggkit-prover-runtime")
-        .enable_all()
-        .build()?;
-
-    let metrics_runtime = tokio::runtime::Builder::new_multi_thread()
-        .thread_name("metrics-runtime")
-        .worker_threads(2)
-        .enable_all()
-        .build()?;
-
-    let aggchain_proof_service = prover_runtime.block_on(async {
-        let grpc_service = GrpcService::new(&config.aggchain_proof_service).await?;
-        Ok::<AggchainProofServiceServer<GrpcService>, aggchain_proof_service::Error>(
-            AggchainProofServiceServer::new(grpc_service),
-        )
-    })?;
-
-    _ = ProverEngine::builder()
-        .add_rpc_service(aggchain_proof_service)
-        .add_reflection_service(aggkit_prover_types::FILE_DESCRIPTOR_SET)
-        .set_rpc_runtime(prover_runtime)
-        .set_metrics_runtime(metrics_runtime)
-        .set_cancellation_token(global_cancellation_token)
-        .start();
+    dotenvy::dotenv().ok();
+
+    let cli = aggkit_prover::cli::Cli::parse();
+
+    match cli.cmd {
+        aggkit_prover::cli::Commands::Run { config_path } => {
+            aggkit_prover::runtime(config_path, &version())?
+        }
+        aggkit_prover::cli::Commands::Config => {
+            println!(
+                "{}",
+                toml::to_string_pretty(&aggkit_prover_config::ProverConfig::default())
+                    .context("Failed to serialize ProverConfig to TOML")?
+            );
+        }
+        aggkit_prover::cli::Commands::ValidateConfig { config_path } => {
+            match aggkit_prover_config::ProverConfig::try_load(config_path.as_path()) {
+                Ok(config) => {
+                    println!(
+                        "{}",
+                        toml::to_string_pretty(&config)
+                            .context("Failed to serialize ProverConfig to TOML")?
+                    );
+                }
+                Err(error) => eprintln!("{}", error),
+            }
+        }
+        aggkit_prover::cli::Commands::Vkey => {
+            let vkey =
+                prover_executor::Executor::get_vkey(aggchain_proof_service::AGGCHAIN_PROOF_ELF);
+
+            println!(
+                "aggchain_proof_vkey: 0x{}",
+                hex::encode(words_to_bytes_le(&vkey.hash_u32()))
+            );
```

Comment:
Nit: I think the `println!`s are a bit nicer with variables:
```suggestion
            let vkey_hex = hex::encode(words_to_bytes_le(&vkey.hash_u32()));
            println!("aggchain_proof_vkey: 0x{vkey_hex}");
```

There are two more above.